src/Umbraco.Cms.Integrations.Search.Algolia/Notifications/NewContentPublishedHandler.cs

```diff
@@ -0,0 +1,64 @@
+﻿using Microsoft.Extensions.Logging;
+
+using System.Text.Json;
+
+using Umbraco.Cms.Core.Events;
+using Umbraco.Cms.Core.Notifications;
+using Umbraco.Cms.Integrations.Search.Algolia.Builders;
+using Umbraco.Cms.Integrations.Search.Algolia.Migrations;
+using Umbraco.Cms.Integrations.Search.Algolia.Models;
+using Umbraco.Cms.Integrations.Search.Algolia.Services;
+
+namespace Umbraco.Cms.Integrations.Search.Algolia.Notifications
+{
+    public class NewContentPublishedHandler : INotificationHandler<ContentPublishedNotification>
```

Comment:
We'll need a handler on deleting and unpublishing content too, so we can remove the records from the index.